### PR TITLE
Remove Julia 0.6 support

### PR DESCRIPTION
**Pull request by fchorney**
- Raise minimum Julia version to 0.7
- Replace 0.6 on CI with 1.1
- Modernize documentation setup
- Remove dependency on Compat

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,7 +1 @@
-julia 0.6
-AWSCore 0.3
```
Comment (ararslan) [inline REQUIRE]: Direct dependencies still need to be listed in REQUIRE; it's used for constructing the registry.

Comment (fchorney) [reply REQUIRE]: ah ok, updated

File: REQUIRE [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-AWSSDK 0.2.0
+julia 0.7
+AWSCore 0.5.5
+AWSSDK 0.4.0
 BSON
-Compat 0.69.0
-Memento 0.7.0
+Compat 1.5.1
```
Comment (ararslan) [inline REQUIRE]: We don't need Compat here anymore, do we?

Comment (fchorney) [reply REQUIRE]: ah yes you are correct. I guess the fact that I forgot a few of those compat lines was making it yell at me for not having it in.

**Pull request merged**